blocks/amp-timeago/index.js

```diff
@@ -0,0 +1,170 @@
+/* global moment */
+
+/**
+ * Internal block libraries.
+ */
+const { __ } = wp.i18n;
+const {
+	registerBlockType
+} = wp.blocks;
+const {
+	InspectorControls,
+	BlockAlignmentToolbar,
+	BlockControls
+} = wp.editor;
+const {
+	DateTimePicker,
+	PanelBody,
+	TextControl,
+	SelectControl
+} = wp.components;
+import timeago from 'timeago.js';
+
+/**
+ * Register block.
+ */
+export default registerBlockType(
+	'amp/amp-timeago',
+	{
+		title: __( 'AMP Timeago' ),
+		category: 'common',
+		icon: 'backup',
+		keywords: [
+			__( 'Time difference' ),
+			__( 'Time ago' ),
+			__( 'Date' )
+		],
+
+		attributes: {
+			align: {
+				type: 'string'
```

Comment:
I think the attributes should be sourced from the children element attributes, as done in `amp-fit-text`: https://github.com/Automattic/amp-wp/pull/1151/commits/c3d99155004459fce3a97bdff127dded2c66f7e5

This prevents the data from being duplicated between the block comment attributes and the content of the block. This means that if you edit the HTML you don't have to make sure that the two sets of attributes are in sync.

Reply:
Please do the same for `amp-mathml` and any other blocks.

Reply:
Good point, added the sources, currently within this PR, however, not sure if this PR will get merged anytime soon, let me know if it would be better to add in a separate PR.